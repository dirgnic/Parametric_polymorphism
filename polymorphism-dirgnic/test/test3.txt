def identity[A](x : A) {
  return x;
}

def main() {
  let a = identity[Int](5);
  let b = identity[Bool](false);
  return 0;
}
