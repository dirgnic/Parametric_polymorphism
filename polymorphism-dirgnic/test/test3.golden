=== After Monomorphization ===
Def "main" [] [] (Let "a" (App (Var "identity_TInt") (IntLit 5)) (Let "b" (App (Var "identity_TBool") (BoolLit False)) (IntLit 0)))
Def "identity_TInt" [] [("x",TInt)] (Var "x")
Def "identity_TBool" [] [("x",TBool)] (Var "x")
Num 0
